src/api/Base.js

```diff
@@ -170,11 +170,13 @@ class Base {
      * Generic success handler
      *
      * @param {Object} data - The response data
+     * @returns {Promise|null|undefined}
      */
-    successHandler = (data: any): void => {
+    successHandler = (data: any): Promise<any> | null | typeof undefined => {
```

Comment:
This change to have the success handler return a promise seems not ideal to me. It seems like we should adhere to a callback pattern or a Promise based pattern rather than a hybrid of the two. Is this change necessary?

Reply:
Yeah, I agree that it's not ideal. `getContactsPromise()` in `ContactsField` (https://github.com/box/box-ui-elements/blob/master/src/features/unified-share-modal/ContactsField.js#L114) uses Promises, but the existing flow uses callbacks. To get the response to show up properly in `ContactsField`, I returned the result of `successHandler()`. Do you know if there are other ways to handle this case?

Reply:
Could we do something like what the [Feed.js](https://github.com/box/box-ui-elements/blob/4858ac4bd28c0f7a49b547f300f8145a9b5c01f3/src/api/Feed.js#L1261) does? If `updatedGetContactsFn` needs to return a promise, it could create a `new Promise(...)` and pass that `resolve` into the success and error callbacks.